presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowFlightClientHandler.java

```diff
@@ -57,6 +59,10 @@ private ArrowFlightClient initializeClient(Optional<String> uri)
                 }
             }
 
+            if (null == allocator) {
+                allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
do we need to handle multi threaded scenario

Reply:
added the synchronised block